src/renderer/dx/DxRenderer.cpp

```diff
@@ -154,18 +152,30 @@ DxEngine::~DxEngine()
         D3D_FEATURE_LEVEL_9_1,
     };
 

```

Comment:
Leave a comment here explaining what you're doing and why
```suggestion
// Trying hardware first for maximum performance, then trying WARP (software) renderer second in case we're running inside a downlevel VM where hardware passthrough isn't enabled like for Windows 7 in a VM.
```